hera_cal/abscal.py

```diff
@@ -3693,6 +3693,232 @@ def post_redcal_abscal_run(data_file, redcal_file, model_files, raw_auto_file=No
+        # dummy refant. Could be completely flagged for all we know.
+        refant_init = str(hdd.ant_1_array[0])
+    else:
+        refant_init = str(refant)
+    # initialize HERACal to store new cal solutions
+    hc = UVCal()
+    hc = hc.initialize_from_uvdata(uvdata=hdd, gain_convention='divide', cal_style='sky',
+                                   ref_antenna_name=refant_init, sky_catalog=f'{model_file}',
+                                   metadata_only=False, sky_field=field_str, cal_type='gain',
+                                   future_array_shapes=False)
+    hc = io.to_HERACal(hc)
+    hc.update(flags=data_ant_flags)
+    # generate cal object from model to hold model flags.
+    hcm = UVCal()
+    hcm = hcm.initialize_from_uvdata(uvdata=hdm, gain_convention='divide', cal_style='sky',
+                                     ref_antenna_name=refant_init, sky_catalog=f'{model_file}',
+                                     metadata_only=False, sky_field=field_str, cal_type='gain',
+                                     future_array_shapes=False)
+
+    hcm = io.to_HERACal(hcm)
+    hcm.update(flags=model_ant_flags)
+    # init all gains to unity.
+    hcm.gain_array[:] = 1. + 0.j
+    hc.gain_array[:] = 1. + 0.j
+    # set calibration flags to be or of data and model flags
+    hc.flag_array = hc.flag_array | hcm.flag_array
+
+    abscal_gains, abscal_flags, _, _ = hc.build_calcontainers()
+
+    # load in weights if supplied.
+    if auto_file is not None:
+
+        hda = io.to_HERAData(auto_file, filetype="uvh5")
+        bls = [ap for ap in hda.get_antpairs() if ap[0] == ap[1]]
+        auto_data, auto_flags, auto_nsamples = hda.read(bls=bls)
+        # generate wgts from autocorrelations
+        wgts = build_data_wgts(data_flags=data_flags, data_nsamples=data_nsamples, model_flags=model_flags,
+                               autocorrs=auto_data, auto_flags=auto_flags)
+    else:
+        # if no autocorrelation file supplied, use nsamples weights times
+        # or of data and model flags
+        wgts = DataContainer({k: (~data_flags[k]).astype(np.float64)
+                             * (~model_flags[k]).astype(np.float64)
+                             * data_nsamples[k] for k in data_flags})
+
+    # select refant if not specified.
+    if refant is None:
+        refant = pick_reference_antenna(abscal_gains, abscal_flags, hc.freqs, per_pol=True)
+    hc.ref_antenna_name = str(refant)
+    hcm.ref_antenna_name = str(refant)
+
+    my_abscal = AbsCal(data=data, model=model, wgts=wgts, freqs=data.freqs)
+
+    # initialize convergence metric to be very large. This will always be greater then
+    # all non-infinite tols provided so calibration loop will run at least once.
+    delta = np.inf
+
+    abscal_gains_iteration = {k: np.ones((hc.Ntimes, hc.Nfreqs), dtype=complex) for k in abscal_gains}
+
+    # find initial starting point with lincal before performing abscal.
+    if dly_lincal:
+        my_abscal.delay_lincal()
```

Comment:
good to expose kword arguments specifically for delay_lincal, they can have a profound effect on the calibration solution on real data.

Perhaps something like

dlycal_kwargs={} in the signature and then

```
my_abscal.delay_lincal(**dlycal_kwargs)
```

Reply:
exposed these.